MSI-2882: Store Pickup button present on check-out when products are …

novikor opened the pull request:
<!---
    Thank you for contributing to Magento Inventory.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
Check if In-Store Pickup delivery is applicable for the cart to decide if the switcher button should be rendered.

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/inventory#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. magento/inventory#2882: Store Pickup button present on check-out when products are not not available for pickup

### Manual testing scenarios (*)
See #2882


### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

novikor committed:
MSI-2882: Store Pickup button present on check-out when products are not not available for pickup.
Check if In-Store Pickup delivery is applicable for the cart to decide if switcher button should be rendered.

swnsma commented on InventoryInStorePickupShippingApi/Model/IsInStorePickupDeliveryAvailableForCart.php:
```diff
@@ -0,0 +1,123 @@
+ *
+ * @api
+ */
+class IsInStorePickupDeliveryAvailableForCart
```
Could you move implementation into **InventoryInStorePickupShipping** and here only leave an interface.

sidolov replied:
I'm not sure that we need interface here. API model looks good as for me

novikor replied:
Well, I will create an interface while it is not too late.

novikor replied:
Done.

swnsma commented on InventoryInStorePickupFrontend/Block/Checkout/LayoutProcessor.php:
```diff
@@ -112,12 +113,6 @@ private function addStorePickupComponents(array $jsLayout): array
-        }
-
-        return (float)$this->config->getValue(self::SEARCH_RADIUS, ScopeInterface::SCOPE_WEBSITE, $website->getId());
+        return (float)$this->config->getValue(self::SEARCH_RADIUS);
```
It will took only value in default scope.
Does it expected behaviour?

novikor replied:
No, it is not

swnsma commented on InventoryInStorePickupShippingApi/Model/IsInStorePickupDeliveryAvailableForCart.php:
```diff
@@ -0,0 +1,123 @@
+
+            return false;
+        } catch (NoSuchEntityException $e) {
+            return false;
```
Could you leave only one 'return' statement ?

novikor replied:
Is it better to have an empty `catch` statement instead of 2 `return`s placed close to each other?
I am not sure about that

swnsma replied:
Just have `$result = false` in the beginning, `$result = true` in the 'if' and `$result = false` in catch. Then - return.

novikor replied:
Genius

swnsma commented on InventoryInStorePickupShippingApi/Model/IsInStorePickupDeliveryAvailableForCart.php:
```diff
@@ -0,0 +1,123 @@
+    /**
+     * Return default country code
+     *
+     * @return string
```
It also nullable, please add 'null' to the doc.

swnsma reviewed (commented):
As PR still in 'Draft', I have added some food for thoughts.

sidolov reviewed (changes requested):
Please, review comments from @swnsma

novikor committed:
Merge branch '1.2-develop' into msi-2882

novikor committed:
MSI-2882: Store Pickup button present on check-out when products are not not available for pickup.
CR and static tests changes.

novikor committed:
Merge remote-tracking branch 'origin/1.2-develop' into msi-2882

swnsma committed:
Merge branch '1.2-develop' into msi-2882

swnsma committed:
Merge branch '1.2-develop' into msi-2882

novikor committed:
MSI-2882: Store Pickup button present on check-out when products are not not available for pickup.
Upgraded API model to API interface implementation.

novikor committed:
Merge remote-tracking branch 'origin/msi-2882' into msi-2882

swnsma commented on InventoryInStorePickupShippingApi/Api/IsInStorePickupDeliveryAvailableForCartInterface.php:
```diff
@@ -0,0 +1,25 @@
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickupShippingApi\Api;
```
Maybe put it into  `Magento\InventoryInStorePickupShippingApi\Model` ?
As we already have `Magento\InventoryInStorePickupShippingApi\Model\IsInStorePickupDeliveryCartInterface`

swnsma committed:
Merge branch '1.2-develop' into msi-2882

novikor committed:
MSI-2882: Store Pickup button present on check-out when products are not not available for pickup.
Got rid of deprecated estimateByAddress method usage.

novikor committed:
MSI-2882: Store Pickup button present on check-out when products are not not available for pickup.
Moved API interface to the model namespace;
Moved implementation to the implementation module.

novikor committed:
MSI-2882: Store Pickup button present on check-out when products are not not available for pickup.
Integration tests coverage.

swnsma reviewed (approved):
Failed unit test is not related to current Pull Request.

sidolov committed:
Merge branch '1.2-develop' into msi-2882
